localstack/services/stepfunctions/asl/component/state/state_execution/state_task/lambda_eval_utils.py

```diff
@@ -19,13 +19,25 @@ def __init__(self, function_error: Optional[str], payload: str):
         self.payload = payload
 
 
+def _from_payload(payload_streaming_body: IO[bytes]) -> Union[json, str]:
+    payload_bytes: bytes = payload_streaming_body.read()
+    decoded_data: str = payload_bytes.decode("utf-8")
+    try:
+        json_data: json = json.loads(decoded_data)
+        return json_data
+    except (UnicodeDecodeError, json.JSONDecodeError):
+        return decoded_data
+
+
 def exec_lambda_function(env: Environment, parameters: dict, region: str, account: str) -> None:
     lambda_client = boto_client_for(region=region, account=account, service="lambda")
 
     invocation_resp: InvocationResponse = lambda_client.invoke(**parameters)
 
     func_error: Optional[str] = invocation_resp.get("FunctionError")
-    payload_json = json.load(invocation_resp["Payload"])
```

Comment:
ups, did I break something with [this refactoring](https://github.com/localstack/localstack/pull/10210/files#diff-37e69a57c7232baa820bebbbfe3afd41f5b8e25fddfbbe6f7e5a5dfed991bb20R28)?
That's a cumbersome decoding step, do we have a regression test to catch that?

Reply:
maybe a clarifying comment on why this extra step is needed could help. Is this payload extraction handling the decoding error necessary in other places of LocalStack as well 🤔 ?

Reply:
Thank you! I have added some in-code comments to explain the goal of the _from_payload function. The lambda tests added by this pr will control the case in which the output of the lambda is a to be handled by SFN as a string and not a json string.